### DISCUSSION TITLE
tar_pin() [ideas]

### DISCUSSION THREAD

**Discussion by Aariq** [ropensci/tarchetypes#147] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I'm using the `pins` package to store a dataset that is at the start of a targets pipeline.  It would be great to have a shortcut to have that target invalidate automatically if the pin hash changes.  I'm imagining something like:

``` r
tar_pin(mydata_raw, board = board_connect(),  name = "ericrscott/mydata")
```

Which, under the hood, uses `pin_read(board = board_connect(), name = "ericrscott/mydata")` to read in the data and uses `pin_meta(board = board_connect(), name = "ericrscott/setaria_runs")$pin_hash` to get a hash to figure out if the target needs to be invalidated.

I guess this would be implemented as a target factory where a target of the pin hash gets created upstream from the `pin_read()` bit?

**Comment by wlandau** [comment]
I would be open to a PR for this. In your case, maybe `format = "url"` and/or `tar_download()` could help if you can get the direct downloadable URL of the dataset.

When I have used `pins` in `targets` pipelines, those pin targets were usually as far as possible downstream, and I trusted the pinboards to stay put. So hashes were not as much of a concern there.
